### PR TITLE
Add testids

### LINKED ISSUE #618: Would like to have some data-testid attributes on ui components of the datatable
Hi, I have a mui-datatable that has some custom search functionality that I'd like to test. I've gotten really into the react-testing-library approach of actually testing the component, and am running into issues with being able to select different elements of the datatable. If you'd be okay with it, I'd like to submit a PR to add `data-testid` props to key elements of the component.

### PR DESCRIPTION
**Pull request by jdetle**
Closes #618

### CODE REVIEW

File: src/components/TableBody.js [hunk 1]
```diff
@@ -159,6 +160,7 @@ class TableBody extends React.Component {
                           {...(columns[columnIndex].setCellProps
                             ? columns[columnIndex].setCellProps(column, dataIndex, columnIndex)
                             : {})}
+                          data-testid={`${columnIndex}-${rowIndex}-MuiDataTableBodyCell`}
```
Comment (gabrielliwerant) [inline src/components/TableBody.js]: Is there a reason to put the indexes in front of the name here? Putting the name first makes parsing into it easier, e.g. `startsWith`.

**Comment by jdetle** [comment]
Totally my bad, will fix that.

**Comment by gabrielliwerant** [comment]
Continued here https://github.com/gregnb/mui-datatables/pull/802

**Comment by gabrielliwerant** [comment]
Should now be available in `2.7.0`.

**Pull request closed without merging**